zebrad/src/components/metrics.rs

```diff
@@ -19,6 +19,28 @@ impl MetricsEndpoint {
             match endpoint_result {
                 Ok(()) => {
                     info!("Opened metrics endpoint at {}", addr);
+
+                    // Expose binary metadata to metrics, using a single time series with
+                    // value 1:
+                    //     https://www.robustperception.io/exposing-the-software-version-to-prometheus
+                    //
+                    // We manually expand the metrics::increment!() macro because it only
+                    // supports string literals for metrics names, preventing us from
+                    // using concat!() to build the name.
+                    static METRIC_NAME: [metrics::SharedString; 2] = [
+                        metrics::SharedString::const_str(env!("CARGO_PKG_NAME")),
+                        metrics::SharedString::const_str("build.info"),
+                    ];
+                    static METRIC_LABELS: [metrics::Label; 1] =
+                        [metrics::Label::from_static_parts(
+                            "version",
+                            env!("CARGO_PKG_VERSION"),
+                        )];
+                    static METRIC_KEY: metrics::KeyData =
+                        metrics::KeyData::from_static_parts(&METRIC_NAME, &METRIC_LABELS);
+                    if let Some(recorder) = metrics::try_recorder() {
+                        recorder.increment_counter(metrics::Key::Borrowed(&METRIC_KEY), 1);
+                    }
```

Comment:
Nit: Can we move this code into a separate method, and call it *before* we open the endpoint listener?
That way, we're guaranteed that the first scrape has the software version - just in case that ever matters.

It feels a bit weird to add a metric in the metrics component, rather than in the application itself. But I don't really have a better suggestion here.

Reply:
Any metrics calls made before the metrics recorder is loaded will never be captured. I picked here to add the information because it's the earliest point at which we could do so; if there's a better place later on I can move it.

Reply:
Put another way, the endpoint installer is doing two things: setting up the metrics recorder, and starting up the endpoint. The build info is only ever "called" once, and this needs to be after the first part.

Reply:
Ok, that makes sense.

I won't block this PR on extracting the build info into its own function.